myapp/src/pages/Discussion/CreateBoard.js

```diff
@@ -0,0 +1,150 @@
+import React, {useState} from "react";
+import axios from "axios";
+//import {Redirect} from "react-router-dom";
+import Padding from "../../components/Padding";
+import {Form as BootStrapForm, Col} from "react-bootstrap";
+import {NewCard} from "./BoardElements.js";
+
+function CreateBoard(props) {
+  const [board, setBoard] = useState(
+    {
+      groupName: "",
+      name: "",
+      description: ""
+    }
+  );
+
+  function handleChange(event) {
+    const {name, value} = event.target;
+    if (name == "group") {
+      setBoard(
+        {
+          groupName: value,
+          name: board.name,
+          description: board.description
+        }
+      );
+    }
+    if (name == "name") {
+      setBoard(
+        {
+          groupName: board.groupName,
+          name: value,
+          description: board.description
+        }
+      );
+    }
+    if (name == "description") {
+      setBoard(
+        {
+          groupName: board.groupName,
+          name: board.name,
+          description: value
+        }
+      );
+    }
+  }
+
+  async function makePostCall(thread) {
+    try {
+      const response = await axios.post('http://localhost:5000/discussion', thread);
+      return response;
+    }
+    catch (error) {
+      console.log(error);
+      return false;
+    }
+  }
+
+  function postThread(thread) {
```

Comment:
To condense this, you could create an array that holds the fields that need to be checked then map through the array checking if it's empty